string path
---
string debug_info
string detection_info
